libs/devices/api-device/feature/src/lib/api/delete-entities.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import {
+  ApiDevice,
+  DeleteEntitiesRequest,
+  DeleteEntitiesResponse207,
+} from "devices/api-device/models"
+import { ApiDeviceSerialPort, Response } from "devices/api-device/adapters"
+
+const CHUNK_SIZE = 500
+
+const createChunks = <T>(items: T[], size: number) => {
+  const chunkCount = Math.ceil(items.length / size)
+  return Array.from({ length: chunkCount }, (_unused, index) =>
+    items.slice(index * size, (index + 1) * size)
+  )
+}
+
+type DeleteEndpoint = "ENTITIES_DATA"
+type DeleteMethod = "DELETE"
+type DeleteEntitiesResponse = Response<DeleteEndpoint, DeleteMethod>
+
+function hasFailedIds(value: unknown): value is { failedIds: string[] } {
+  if (value === null || typeof value !== "object") return false
+  const candidate = value as { failedIds?: unknown }
+  return (
+    Array.isArray(candidate.failedIds) &&
+    candidate.failedIds.every((id) => typeof id === "string")
+  )
+}
```

Comment:
[Nitpick] Lodash could be helpful here too


```suggestion
function hasFailedIds(value: unknown): value is { failedIds: string[] } {
  return (
    isObject(value) &&
    "failedIds" in value &&
    isArray(value.failedIds) &&
    value.failedIds.every(isString)
  )
}
```